packages/dart/test/hub_span_test.dart

```diff
@@ -0,0 +1,311 @@
+
+          expect(span.name, equals('my-span-name'));
+        });
+
+        test('sets attributes on span when provided', () {
+          final hub = fixture.getSut();
+          final attributes = {
+            'attr1': SentryAttribute.string('value1'),
+            'attr2': SentryAttribute.int(42),
+          };
+
+          final span = hub.startSpan('test-span', attributes: attributes);
+
+          expect(span.attributes, equals(attributes));
+        });
+      });
+
+      group('active span handling', () {
+        test('sets span as active on scope when active is true', () {
+          final hub = fixture.getSut();
+
+          final span = hub.startSpan('test-span', active: true);
+
+          expect(hub.scope.getActiveSpan(), equals(span));
+        });
+
+        test('does not set span as active on scope when active is false', () {
+          final hub = fixture.getSut();
+
+          hub.startSpan('test-span', active: false);
+
+          expect(hub.scope.getActiveSpan(), isNull);
+        });
+      });
+
+      group('parent span resolution', () {
+        test('creates root span when no active span exists', () {
+          final hub = fixture.getSut();
+
+          final span = hub.startSpan('test-span');
+
+          expect(span.parentSpan, isNull);
+        });
+
+        test('uses active span as parent when parentSpan is not provided', () {
+          final hub = fixture.getSut();
+          final parentSpan = hub.startSpan('parent-span');
+
+          final childSpan = hub.startSpan('child-span');
+
+          expect(childSpan.parentSpan, equals(parentSpan));
+        });
+
+        test('uses explicit parentSpan instead of active span when provided',
+            () {
+          final hub = fixture.getSut();
+          final explicitParent = hub.startSpan('explicit-parent');
+          hub.startSpan('other-span'); // Changes active span
+
+          final childSpan = hub.startSpan(
+            'child-span',
+            parentSpan: explicitParent,
+            active: false,
+          );
+
+          expect(childSpan.parentSpan, equals(explicitParent));
+        });
+
+        test('allows finished span to be used as parent', () {
+          final hub = fixture.getSut();
+          final finishedParent = hub.startSpan('finished-parent');
+          finishedParent.end();
+
+          final childSpan = hub.startSpan(
+            'child-span',
+            parentSpan: finishedParent,
+          );
+
+          expect(childSpan.parentSpan, equals(finishedParent));
+        });
+
+        test('creates root span when parentSpan is explicitly set to null', () {
+          final hub = fixture.getSut();
+          hub.startSpan('active-span');
+
+          final rootSpan = hub.startSpan('root-span', parentSpan: null);
+
+          expect(rootSpan.parentSpan, isNull);
+        });
+
+        test('should not allow finished span to be use as parent', () {
+          // TODO: this test case needs more clarification
```

Comment:
Will this be done in this PR?

Reply:
Sorry this is left-over, we will allow finished spans to be set as parent and send them - we'll let downstream processing handle it